fontbe/src/gvar.rs

```diff
@@ -0,0 +1,114 @@
+//! Generates a [gvar](https://learn.microsoft.com/en-us/typography/opentype/spec/gvar) table.
+
+use log::trace;
+
+use font_types::{F2Dot14, GlyphId};
+use fontdrasil::orchestration::Work;
+use fontir::variations::VariationRegion;
+use write_fonts::{
+    dump_table,
+    tables::{
+        gvar::{GlyphDeltas, GlyphVariations, Gvar},
+        variations::Tuple,
+    },
+};
+
+use crate::{
+    error::Error,
+    orchestration::{BeWork, Bytes, Context, GvarFragment},
+};
+
+struct GvarWork {}
+
+pub fn create_gvar_work() -> Box<BeWork> {
+    Box::new(GvarWork {})
+}
+
+/// <https://learn.microsoft.com/en-us/typography/opentype/spec/otvaroverview#variation-data>
+#[derive(Debug, Default)]
+struct TupleBuilder {
+    axis_names: Vec<String>,
+    min: Vec<F2Dot14>,
+    peak: Vec<F2Dot14>,
+    max: Vec<F2Dot14>,
+}
+
+impl TupleBuilder {
+    fn build(self) -> (Tuple, Tuple, Tuple) {
+        (
+            Tuple::new(self.min),
+            Tuple::new(self.peak),
+            Tuple::new(self.max),
+        )
+    }
+}
+
+impl From<&VariationRegion> for TupleBuilder {
+    fn from(region: &VariationRegion) -> Self {
+        let mut builder = TupleBuilder::default();
+        for (axis_name, tent) in region.iter() {
+            builder.axis_names.push(axis_name.clone());
+            builder.min.push(F2Dot14::from_f32(tent.lower.to_f32()));
+            builder.peak.push(F2Dot14::from_f32(tent.peak.to_f32()));
+            builder.max.push(F2Dot14::from_f32(tent.upper.to_f32()));
+        }
+        trace!("{builder:?}");
+        builder
+    }
+}
+
+fn to_deltas(fragment: &GvarFragment) -> Vec<GlyphDeltas> {
+    fragment
+        .deltas
+        .iter()
+        .filter_map(|(region, deltas)| {
+            if region.is_default() {
+                return None;
+            }
+
+            // Variation of no point has limited entertainment value
+            if deltas.is_empty() {
+                return None;
+            }
+
+            // TODO: nice rounding on deltas
+            let deltas: Vec<_> = deltas.iter().map(|v| (v.x as i16, v.y as i16)).collect();
+
+            let tuple_builder: TupleBuilder = region.into();
+            let (min, peak, max) = tuple_builder.build();
+            Some(GlyphDeltas::new(peak, deltas, Some((min, max))))
```

Comment:
if we are always passing the full min/max/peak then maybe I should change this API?

Reply:
Yes please, I managed to confuse myself about whether that was the "normal" min/peak/max or not.